src/components/accession2/CalculatorModal.tsx

```diff
@@ -0,0 +1,138 @@
+  open: boolean;
+  record: Accession2;
+  setRecord: React.Dispatch<React.SetStateAction<Accession2>>;
+  onClose: () => void;
+  reload: () => void;
+  organization: ServerOrganization;
+  onChange: (id: string, value: unknown) => void;
+  openPreviousModal: () => void;
+}
+
+export default function CalculatorDialog(props: CalculatorDialogProps): JSX.Element {
+  const { onClose, open, record, onChange, setRecord, openPreviousModal } = props;
+
+  const snackbar = useSnackbar();
+
+  const getTotalCount = async () => {
+    const response = await updateAccession2(record);
+    if (response.requestSucceeded && response.accession) {
+      goToPrev();
+      setRecord(response.accession);
+    } else {
+      snackbar.toastError();
+    }
+  };
+
+  const onCloseHandler = () => {
+    onClose();
+  };
+
+  const goToPrev = () => {
+    openPreviousModal();
+    onCloseHandler();
+  };
+
+  const onChangeSubsetWeight = (value: number) => {
+    setRecord({
+      ...record,
+      subsetWeight: { quantity: value, units: record.subsetWeight?.units || 'Grams' },
+    });
+  };
+
+  const onChangeSubsetUnit = (newValue: Unit) => {
+    setRecord({
+      ...record,
+      subsetWeight: { quantity: record.subsetWeight?.quantity || 0, units: newValue.value },
+    });
+  };
+
+  const onChangeRemainingQuantity = (value: number) => {
+    setRecord({
+      ...record,
+      remainingQuantity: { quantity: value, units: record.remainingQuantity?.units || 'Grams' },
+    });
+  };
+
+  const onChangeRemainingQuantityUnit = (newValue: Unit) => {
+    if (record) {
+      setRecord({
+        ...record,
+        remainingQuantity: { quantity: record.remainingQuantity?.quantity || 0, units: newValue.value },
+      });
+    }
+  };
+
+  return (
+    <>
+      <DialogBox
+        onClose={onCloseHandler}
+        open={open}
+        title={strings.QUANTITY}
+        size='medium'
+        middleButtons={[
+          <Button label={strings.BACK} type='passive' onClick={goToPrev} priority='secondary' key='button-1' />,
+          <Button onClick={getTotalCount} label={strings.GET_TOTAL_COUNT} key='button-2' />,
+        ]}
+      >
+        <Grid item xs={12} textAlign='left'>
+          <Textfield
+            label={strings.SUBSET_WEIGHT}
+            id='subsetWeight'
+            onChange={(id, value) => onChangeSubsetWeight(value as number)}
+            type='text'
+            value={record.subsetWeight?.quantity}
+          />
+          <SelectT<Unit>
+            options={WEIGHT_UNITS_V2}
+            placeholder={strings.SELECT}
+            onChange={onChangeSubsetUnit}
+            isEqual={(a: Unit, b: Unit) => a.value === b.value}
+            renderOption={(unit) => unit.label}
+            displayLabel={(unit) => (unit ? unit.label : 'g')}
+            selectedValue={WEIGHT_UNITS_V2.find((wu) => wu.value === record.subsetWeight?.units)}
+            toT={(label: string) => ({ label } as Unit)}
+            fullWidth={true}
+          />
```

Comment:
in my most recent PR I created a new dropdown utility, it will be easier to use for later.

Reply:
i merged my SelectValue component if we want to use that

Reply:
Yes, will do!

Reply:
Replace these, I will rename to Dropdown later as part of my PR https://github.com/terraware/terraware-web/pull/496.. which I won't merge until after this PR.